### PR TITLE
fix CORS error

### PR DESCRIPTION
**Pull request by Maheima**
Solves https://github.com/Real-Dev-Squad/website-backend/issues/756
Changed the Regex from `/(\.realdevsquad\.com$)|(localhost)/` to `/(https:\/\/([a-zA-Z0-9-_]+\.)?realdevsquad\.com$)|(localhost)/` in Allowed origins

### CODE REVIEW

File: config/development.js [hunk 1]
```diff
@@ -25,7 +26,7 @@ module.exports = {
   },
 
   cors: {
-    allowedOrigins: /(\.realdevsquad\.com$)|(localhost)/, // Allow *.realdevsquad.com and localhost for non-production envs
+    allowedOrigins: /(https:\/\/(.*\.)*realdevsquad\.com$)|(localhost)/, // Allow *.realdevsquad.com and localhost for non-production envs
```
Comment (ankushdharkar) [inline config/development.js:29]: > Allow *.realdevsquad.com and localhost for non-production envs

The comment is slightly incorrect.

We also allow `realdevsquad.com` (without the subdomain), correct?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: config/development.js (hunk 1 above)
Comment (RitikJaiswal75) [inline config/development.js:29]: currently this will allow a.b.realdevsquad.com, that should not happen I guess please check and fix

Comment (Maheima) [reply config/development.js]: fixed it

Comment (Maheima) [reply config/development.js]: Done

File: config/staging.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ module.exports = {
   },
 
   cors: {
-    allowedOrigins: /(\.realdevsquad\.com$)|(localhost)/, // Allow *.realdevsquad.com and localhost for non-production envs
+    allowedOrigins: /(https:\/\/([a-zA-Z0-9-_]+\.)?realdevsquad\.com$)|(localhost)/, // Allow *.realdevsquad.com and localhost for non-production envs
```
Comment (shubham-y) [inline config/staging.js:20]: Please update the comment here as well as in `config/test.js`

File: config/staging.js [hunk 3]
```diff
@@ -16,10 +16,6 @@ module.exports = {
     },
   },
 
-  cors: {
```
Comment (ankurnarkhede) [inline config/staging.js:19]: We are not overriding the values in env specific config. So it shouldnt be added here... 🤔

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: config/default.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ module.exports = {
   },
 
   cors: {
-    allowedOrigins: /\.realdevsquad\.com$/,
+    allowedOrigins: /(https:\/\/([a-zA-Z0-9-_]+\.)?realdevsquad\.com$)/, // Allow realdevsquad.com, *.realdevsquad.com and localhost for non-production envs
```
Comment (ankurnarkhede) [inline config/default.js:52]: Also, can you add why this issue is occurring now, what has changed?

**Review by ankurnarkhede — DISMISSED** [review]

File: config/staging.js (hunk 3 above)
Comment (ankurnarkhede) [reply config/staging.js:19]: Okay, this has been removed 👍

File: config/default.js (hunk 4 above)
Comment (ankurnarkhede) [reply config/default.js]: As discussed, this is because of www not present in the domain.

**Comment by ankushdharkar** [comment]
🚢

**Review by ankushdharkar — APPROVED** [review]
🚢

**Pull request merged**